### DISCUSSION TITLE
USB webcam redirection in wifi

### DISCUSSION THREAD

**Discussion by kcosmos4** [FreeRDP/FreeRDP#7473] [Q&A] [answered]
I'm testing the USB webcam redirection function.
USB webcam redirection works normally when wired LAN is connected. However, when connected via wifi, an error occurs and the program terminates.

I built it using the freerdp 2.4.1 source.
I installed Debian Linux on VirtualBox and set up a FreeRDP build environment.
I package FreeRDP after build and copy this to a laptop with Debian installed. I install the package and run the test.

The error that occurred went below.
[ERROR] [com.freerdp.core.transport] - BIO_write returned a system error 32: Broken pipe
[ERROR] [com.freerdp.core] - transport_write: freerdp_set_last_error_ex ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]

libfreerdp/core/transport.c - Occurs in the following conditional statement in the transport_write() function.
/* non-blocking can live with blocked IOs */
if (!transport->blocking)
{
WLog_ERR_BIO(transport, "BIO_write", transport->frontBio);
goto out_cleanup;
}

please help what i should do
thank you.

**Comment by akallabeth** [comment, marked as answer]
That error just tells that the connection to the server failed.
(e.g. your network went down or similar)

**Reply by kcosmos4** [reply]
Thank you for answer.

I have additional questions.
Why does this happen only in a wireless LAN environment and not in a wired LAN environment?

Where is the problem in the wireless LAN network environment?

thank you.

**Reply by kcosmos4** [reply]
The wireless LAN network environment I tested used the Internet sharing function on the MacBook Pro. I replaced this with a wireless router device.

Now, when the webcam is connected, the program does not close and works well.
However, performance problems still remain.

After writing an additional question in the comments, your answer was a hint and I was able to solve it.
Thank you very much.

**Reply by akallabeth** [reply]
> Thank you for answer.
> 
> I have additional questions. Why does this happen only in a wireless LAN environment and not in a wired LAN environment?
> 
> Where is the problem in the wireless LAN network environment?
> 
> thank you.

it is wireless, e.g. not reliable (much less than wired) and varying latency

**Reply by akallabeth** [reply]
> The wireless LAN network environment I tested used the Internet sharing function on the MacBook Pro. I replaced this with a wireless router device.
> 
> Now, when the webcam is connected, the program does not close and works well. However, performance problems still remain.
> 
> After writing an additional question in the comments, your answer was a hint and I was able to solve it. Thank you very much.

this is raw USB redirection, so it uses LOTS of bandwidth and does not tolerate latency well.
